### DISCUSSION TITLE
bun install is slower than npm ci

### DISCUSSION THREAD

**Discussion by beeb** [oven-sh/bun#6163] [Q&A]
I normally use `npm ci` to install dependencies into the build step of a Dockerfile.

With my project, this takes about ~4.4s~ 2.7s to 3.2s for the install step, on my machine.

I tried to switch to bun.
Generated a `bun.lockb` file beforehand and copying it inside the container.
Using the following command instead: `bun i --ignore-scripts --frozen-lockfile`, it takes ~22.7s~ 3.3s to 5s.

With `--no-verify` the results are almost identical.

How can I speed this up?

EDIT: repeated on Ubuntu (not using Windows/WSL2)

**Comment by tomerh2001** [comment]
Windows?

**Reply by beeb** [reply]
Docker Desktop is running on Windows but it's using WSL2 under the hood I believe.

**Reply by beeb** [reply]
Repeated the experiment on my Ubuntu laptop (multiple runs in succession without docker caching for each): 

npm ci: 2.7s to 3.2s
bun install (with command from the post above): 3.3s to 5s

I'm glad it's nowhere near as bad, but still is there room for improvement, as I imagine this will become even more apparent in CI?

**Reply by melroy89** [reply]
Could you share a package.json file or something? At least create a reproducible example for us.

**Reply by beeb** [reply]
@melroy89 will start working on creating a repro using a similar setup to what I have here (private CodeArtifact registry for some dependencies)
